cmd/skaffold/app/cmd/dev.go

```diff
@@ -34,7 +34,11 @@ func NewCmdDev() *cobra.Command {
 		WithDescription("Run a pipeline in development mode").
 		WithCommonFlags().
 		WithFlags(func(f *pflag.FlagSet) {
-			f.StringVar(&opts.Trigger, "trigger", "polling", "How are changes detected? (polling, manual or notify)")
+			f.StringVar(&opts.Trigger, "trigger", "polling", "How are changes detected? (polling, notify, or manual)")
+			f.StringVar(&opts.BuildTrigger, "build-trigger", "auto", "How are builds triggered? (auto, manual, or api - auto defers control to the watch trigger)")
```

Comment:
It feels bad to put a lot of options in the face of the final user. I assume those flags will only be used by IDEs.

Reply:
yeah, I'm not sure that they'll be used **only** by the IDEs but thats definitely the primary use case. maybe I should make them hidden flags? not sure where else they belong.

Reply:
I think it's a great idea to make IDE-only flags hidden! +1

Reply:
As discussed change these to AutoBuild/Sync/Deploy, and remove APIMode.